@here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts

```diff
@@ -64,13 +65,104 @@ export class VolatileLayerClient {
+     * @param abortSignal A signal object that allows you to communicate with a request (such as a Fetch)
+     * and abort it if required via an AbortController object
+     *  @see https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal
+     */
+    async getData(
+        dataRequest: DataRequest,
+        abortSignal?: AbortSignal
+    ): Promise<Response> {
+        const dataHandle = dataRequest.getDataHandle();
+        const partitionId = dataRequest.getPartitionId();
+        const quadKey = dataRequest.getQuadKey();
+
+        if (
+            dataHandle !== undefined ||
+            partitionId !== undefined ||
+            quadKey !== undefined
+        ) {
+            if (dataHandle) {
+                return this.downloadPartition(dataHandle, abortSignal);
+            }
+
+            if (partitionId) {
+                const partitionIdDataHandle = await this.getDataHandleByPartitionId(
+                    partitionId
+                ).catch((error: Response) => Promise.reject(error));
+                return this.downloadPartition(
+                    partitionIdDataHandle,
+                    abortSignal
+                );
+            }
+
+            if (quadKey) {
+                const quadKeyDataHandle = await this.getDataHandleByQuadKey(
+                    quadKey
+                ).catch((error: Response) => Promise.reject(error));
+                return this.downloadPartition(quadKeyDataHandle, abortSignal);
+            }
+        }
+
+        return Promise.reject(
+            new Error(
+                `No data provided. Add dataHandle, partitionId or quadKey to the DataRequest object`
+            )
+        );
+    }
+
+    /**
+     * Fetch and returns partition metadata
+     * @param partitionId The name of the partition to fetch.
+     * @returns A promise of partition metadata which used to get partition data
+     */
+    private async getDataHandleByPartitionId(
+        partitionId: string
+    ): Promise<string> {
+        const queryRequestBilder = await this.getRequestBuilder("query");
+        const partitions = await QueryApi.getPartitionsById(
+            queryRequestBilder,
+            {
+                layerId: this.layerId,
+                partition: [partitionId]
+            }
+        );
+        const partition = partitions.partitions.find(element => {
+            return element.partition === partitionId;
+        });
+
+        return partition && partition.dataHandle
+            ? partition.dataHandle
+            : Promise.reject(
+                  `No partition dataHandle for partition ${partitionId}. HRN: ${this.hrn}`
+              );
+    }
+
+    private async getDataHandleByQuadKey(quadKey: QuadKey): Promise<string> {
+        if (!utils.isValid(quadKey)) {
+            return Promise.reject(new Error("QuadKey is not valid"));
+        }
+
+        const index = await this.getIndexFor(quadKey);
```

Comment:
Please, wait for PR #46 will be merged. Instead of getIndexFor we'll be using getPartitions by QuadKey